tests/test_project_time.py

```diff
@@ -21,52 +21,64 @@
 
 
 @pytest.fixture
-def registry_mgr():
+def project():
     conn = DatabaseConnection(database="simple-standard-scenarios")
-    return RegistryManager.load(conn, offline_mode=True)
-
-
-def test_no_unexpected_timezone():
-    for tzo in TimeZone:
-        assert (
-            tzo.is_standard() + tzo.is_prevailing() == 1
-        ), f"{tzo} can either be prevailing or standard"
-
-
-def test_convert_to_project_time(registry_mgr):
+    registry_mgr = RegistryManager.load(conn, offline_mode=True)
     project_id = "dsgrid_conus_2022"
     project = registry_mgr.project_manager.load_project(project_id)
+    return project
+
 
+@pytest.fixture
+def resstock(project):
     dataset_id = "resstock_conus_2022_reference"
     project.load_dataset(dataset_id)
-    resstock = project.get_dataset(dataset_id)
+    return project.get_dataset(dataset_id)
+
+
+@pytest.fixture
+def resstock_inputdatasetmodel(project):
+    dataset_id = "resstock_conus_2022_reference"
+    return project.config.get_dataset(dataset_id)
+
 
+@pytest.fixture
+def comstock(project):
     dataset_id = "comstock_conus_2022_reference"
     project.load_dataset(dataset_id)
-    comstock = project.get_dataset(dataset_id)
+    return project.get_dataset(dataset_id)
+
 
+@pytest.fixture
+def tempo(project):
     dataset_id = "tempo_conus_2022"
     project.load_dataset(dataset_id)
-    tempo = project.get_dataset(dataset_id)
-    tempo_load_data = tempo._handler._load_data
-    tempo_load_data_lookup = tempo._handler._load_data_lookup
-
-    # different ways to access project_time_dim:
-    project_time_dim = project.config.get_base_dimension(
-        DimensionType.TIME
-    )  # or tempo._handler._project_time_dim
-    resstock_time_dim = resstock._handler.config.get_dimension(DimensionType.TIME)
-    comstock_time_dim = comstock._handler.config.get_dimension(DimensionType.TIME)
-    comstock_geo_dim = comstock._handler.config.get_dimension(DimensionType.GEOGRAPHY)
-    tempo_time_dim = tempo._handler.config.get_dimension(DimensionType.TIME)
+    return project.get_dataset(dataset_id)
 
-    # [1] test build_time_dataframe()
+
+def test_no_unexpected_timezone():
+    for tzo in TimeZone:
+        assert (
+            tzo.is_standard() + tzo.is_prevailing() == 1
+        ), f"{tzo} can either be prevailing or standard"
+
+
+def test_build_time_dataframe(project, resstock, comstock):
+    project_time_dim = project.config.get_base_dimension(DimensionType.TIME)
+    # alteratively, project_time_dim = resstock._handler._project_time_dim
```

Comment:
We shouldn’t store that attribute in the handler. It can retrieve the time dim on demand.

Reply:
I don’t care about tests accessing private methods. That’s often the best way to unit-test the code. I am suggesting that the dataset handler doesn’t need to store the project time dimension as an attribute. It feels odd keep that reference. The handler can retrieve that dimension config object from the project_config whenever it needs it. It’s a simple change but doesn’t have to be done here.

Reply:
I think it’s best to remove this comment. It is currently identical to the line above it.